lib/srv/sessiontracker_test.go

```diff
@@ -108,13 +100,20 @@ func TestSessionTracker_UpdateRetry(t *testing.T) {
 		if i%2 == 0 {
 			svc.updateError <- updateError
 
-			// wait for the retry to be engaged
+			// the first retry is instant, wait for the update to be called
+			// again without advancing the clock
+			waitForUpdate(t, svc, done)
+
+			// send another error
+			svc.updateError <- updateError
+
+			// wait for the second retry to be engaged
 			clock.BlockUntil(1)
 
 			// advance far enough for the retry to fire
 			clock.Advance(65 * time.Second)
 
-			// wait for the update to be called again
+			// wait for the update to be called a third time
 			waitForUpdate(t, svc, done)
```

Comment:
Hey @rosstimothy I discovered that the first retry is actually firing instantly, which caused this test to become flaky after updating `clockwork`. For now I've updated the test to work with the current code, but lmk if you think we should change the production code to introduce a delay before the first retry.

Reply:
added an initial wait